Improved error message

adhoniran opened the pull request:
### Fixes: #13956

Improved error handling in the get_feed() method to provide a more descriptive error message when NetBox is unable to connect to the feed server. Instead of returning the original exception, we now return a custom error message for better clarity and security.

Changing the error message when NetBox is unable to connect to feed server to 'NetBox is unable to connect to feed server...', instead to show the error exception variable content

adhoniran committed:
Improved error message
Improved error handling in the get_feed() method to provide a more descriptive error message when NetBox is unable to connect to the feed server. Instead of returning the original exception, we now return a custom error message for better clarity and security.

adhoniran committed:
Improved error message
Improved error handling in the get_feed() method to provide a more descriptive error message when NetBox is unable to connect to the feed server. Instead of returning the original exception, we now return a custom error message for better clarity and security.

adhoniran committed:
Removed exception variable assignment into get_feed()
Removed variable assignment because variable is unused.

jeremystretch commented on netbox/extras/dashboard/widgets.py:
```diff
@@ -318,9 +318,10 @@ def get_feed(self):
             response.raise_for_status()
-        except requests.exceptions.RequestException as e:
+        except requests.exceptions.RequestException:
+            feed_error = 'NetBox is unable to connect to feed server...'
```
This isn't a very helpful error message. While we can clean up the display of the exception, we do want to retain the original error message, which can offer more insight into the underlying problem. For example, "Temporary failure in name resolution" offers a much bigger clue to the problem than "NetBox is unable to connect to feed server."

jeremystretch commented:
@adhoniran are you still interested in completing this PR?

adhoniran commented:
Yes,

I'm sorry, I had a very very complicated week, but I can do it tomorrow!


Em sex., 13 de out. de 2023 às 10:28, Jeremy Stretch <
***@***.***> escreveu:

> @adhoniran <https://github.com/adhoniran> are you still interested in
> completing this PR?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/netbox-community/netbox/pull/13953#issuecomment-1761521899>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJTTWVGGEXTRFZ4ZKH2UZDX7E6ZJANCNFSM6AAAAAA5RCKHEU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

adhoniran committed:
Enhanced error handling for HTTP requests

Improved the error handling logic in HTTP request function. Now, the error message returned upon exceptions is more specific, providing information about the type of the error occurred (Timeout, URL Required, Too Many Redirects, etc.). If the error is associated with the response from the server, the reason for the HTTP error is now included in the error message.

jeremystretch commented on netbox/extras/dashboard/widgets.py:
```diff
@@ -318,9 +318,25 @@ def get_feed(self):
+            if 'response' in locals():
+                feed_error = f'HTTP {response.reason}'
+            else:
+                if isinstance(e, requests.exceptions.Timeout):
+                    feed_error = 'Timeout'
+                elif isinstance(e, requests.exceptions.URLRequired):
+                    feed_error = 'URL Required'
+                elif isinstance(e, requests.exceptions.TooManyRedirects):
+                    feed_error = 'Too Many Redirects'
+                else:
```
This is a classic anti-pattern. Instead of trying to predict all potential reasons for failure, we should simply convey the failure message contained within the exception to the user.

jeremystretch commented:
@adhoniran are you still interested in working on this?

jeremystretch commented:
Closing this for inactivity
